src/object/cli_ec.c

```diff
@@ -2358,6 +2371,46 @@ obj_ec_recov_stripe(struct obj_ec_recov_codec *codec,
 		       buf_src, buf_err);
 }
 
+/** Encode one full stripe, the result parity buffer will be filled. */
+static void
+obj_ec_recov_single_stripe(struct obj_ec_recov_codec *codec,
+			   struct daos_oclass_attr *oca, void *buf_stripe,
+			   uint64_t cell_sz, uint64_t iod_size)
+{
+	unsigned char	*buf_src[OBJ_EC_MAX_K];
+	unsigned char	*buf_err[OBJ_EC_MAX_P];
+	int		src_idx;
+	int		err_idx;
+	int		i;
+
+	src_idx = obj_ec_data_tgt_nr(oca);
+	err_idx = codec->er_nerrs;
+	for (i = src_idx + err_idx - 1; i >= 0; i--) {
+		struct obj_ec_singv_local loc = {0};
+
+		obj_ec_singv_local_sz(iod_size, oca, i, &loc);
+		if (loc.esl_off % cell_sz != 0) {
+			uint32_t shift = cell_sz - loc.esl_off % cell_sz;
+
+			memmove(buf_stripe + loc.esl_off + shift,
+				buf_stripe + loc.esl_off, loc.esl_size);
+			memset(buf_stripe + loc.esl_off, 0, shift);
+			loc.esl_off += shift;
```

Comment:
I did some test and seems better to change obj_ec_singv_local_sz to avoid the movement here, i'll push my commit to your PR. I changed back the iod_size also.
please check if it is good for you